src/line.rs

```diff
@@ -34,6 +34,19 @@ impl<T> Line<T> {
     }
 }
 
+impl<T: PartialOrd> Line<T> {
+    /// Returns the intersection between the sets, if any.
+    pub fn intersection(self, other: Self) -> Option<Self> {
+        if self.contains(&other.start) && !self.contains(&other.end) {
```

Comment:
I would add just an inline comment: `// behind other` before this, and `// after other` before the 2nd condition.